libs/vertexai/langchain_google_vertexai/chat_models.py

```diff
@@ -316,70 +318,88 @@ def _get_client_with_sys_instruction(
 def _parse_response_candidate(
     response_candidate: "Candidate", streaming: bool = False
 ) -> AIMessage:
-    try:
-        content = response_candidate.text
-    except AttributeError:
-        content = ""
-
+    content = None
     additional_kwargs = {}
-    first_part = response_candidate.content.parts[0]
-    if first_part.function_call:
-        function_call = {"name": first_part.function_call.name}
-        # dump to match other function calling llm for now
-        function_call_args_dict = proto.Message.to_dict(first_part.function_call)[
-            "args"
-        ]
-        function_call["arguments"] = json.dumps(
-            {k: function_call_args_dict[k] for k in function_call_args_dict}
-        )
-        additional_kwargs["function_call"] = function_call
-        if streaming:
-            tool_call_chunks = [
-                ToolCallChunk(
-                    name=function_call.get("name"),
-                    args=function_call.get("arguments"),
-                    id=function_call.get("id", str(uuid.uuid4())),
-                    index=function_call.get("index"),
-                )
-            ]
-            return AIMessageChunk(
-                content=content,
-                additional_kwargs=additional_kwargs,
-                tool_call_chunks=tool_call_chunks,
+    tool_calls = []
+    invalid_tool_calls = []
+    tool_call_chunks = []
+
+    for part in response_candidate.content.parts:
+        try:
+            text = part.text
```

Comment:
nits: maybe `if hasattr(part, "text") and part.text: ...` instead

Reply:
This is a `@property` getter, it would not work